src/nvim/main.c

```diff
@@ -1291,15 +1291,22 @@ static void set_window_layout(mparm_T *paramp)
 static void load_plugins(void)
 {
   if (p_lpl) {
+    char_u *rtp_copy = NULL;
+
     // First add all package directories to 'runtimepath', so that their
     // autoload directories can be found.  Only if not done already with a
     // :packloadall command.
+    // Make a copy of 'runtimepath', so that source_runtime does not use the
+    // pack directories.
     if (!did_source_packages) {
+      rtp_copy = vim_strsave(p_rtp);
       add_pack_start_dirs();
     }
 
-    source_runtime((char_u *)"plugin/**/*.vim", DIP_ALL | DIP_NOAFTER);  // NOLINT
+    source_in_path(rtp_copy, (char_u *)"plugin/**/*.vim",  // NOLINT
```

Comment:
Vim checks for `NULL`, we don't.  `vim_strsave` is a very thin wrapper around `xstrdup` for us, mainly handling casting.